src/main/java/fr/insalyon/citi/golo/runtime/FunctionCallSupport.java

```diff
@@ -108,11 +118,15 @@ public static Object fallback(FunctionCallSite callSite, Object[] args) throws T
       Method method = (Method) result;
       checkLocalFunctionCallFromSameModuleAugmentation(method, callerClass.getName());
       types = method.getParameterTypes();
-      if (method.isVarArgs() && isLastArgumentAnArray(types.length, args)) {
+      //TODO improve varargs support on named arguments. Matching the last param type + according argument
```

Comment:
to get inspiration from Python :wink:
given
```python
def f(a, b, c, *args):
    print("got {}, {}, {}, {}".format(a, b, c, args))
```

that is 3 named/positional parameters, and a vararg.

the following calls are valid:
```python
f(1, 2, 3)
f(b=2, a=1, c=3)
f(1, c=3, b=2)
f(1, 2, 3, 4, 5, 6)
```
1. call without varargs with positional arguments: outputs `got 1, 2, 3, ()`
2. call without varargs with named arguments: outputs `got 1, 2, 3, ()`
3. call without varargs, `a` is positional and `b` and `c` named: `got 1, 2, 3, ()`
4. call with positional arguments and varargs: outputs `got 1, 2, 3, (4, 5, 6)`

But the calls
```python
f(a=1, 2, 3)
f(b=2, a=1, c=3, 4, 5, 6)
```
fails. Indeed, once one of the arguments is named, all the *following* arguments *must* be named. So in the first call, even if  `a` is at the “right” place, the other arguments must be named. Compare with 3 above.

It is thus possible to mix named arguments and varargs, but *not* in the same call, since a vararg arguments *must* follows all positional parameters in the function signature, and thus can't follow the named arguments in the call. 
This restriction makes sens to me.

Note also that calling `f(a=1, b=2, c=3, args=(4, 5, 6))` is not valid, since `args` is not actually a keyword parameter for the function, which on the other hand could make sens.

As a side note, the python’s “named varargs” feature could also be realy nice to have:
```python
def g(a, b, *args, **kargs):
    print("got {}, {}, {}, {}".format(a, b, args, kargs))

g(1, 2)
g(c=3, b=2, a=1, d=4)
g(1, 2, c=3)
g(1, 2, 3, 4, 5)
g(1, 2, 3, 4, c=5, d=6)
```

outputs 
```
got 1, 2, (), {}
got 1, 2, (), {'c': 3, 'd': 4}
got 1, 2, (), {'c': 3}
got 1, 2, (3, 4, 5), {}
got 1, 2, (3, 4), {'c': 5, 'd': 6}
```
but here again, all arguments following a named argument must be named.